starfish/image/_segmentation/watershed.py

```diff
@@ -321,6 +321,28 @@ def watershed(self, markers: np.ndarray, watershed_mask: np.ndarray) -> np.ndarr
 
         return res
 
+    @staticmethod
+    def relabel_image(image: np.ndarray) -> np.ndarray:
+        """given a label image where some objects have been removed, relabel it with sequential integers
+
+        Parameters
+        ----------
+        image : np.ndarray[np.uint32]
+            image whose values identify which object each pixel corresponds to. the values may
+            not be sequential integers.
+
+        Returns
+        -------
+        image : np.ndarray[np.uint32]
+            same as input, but the values are re-labled as sequential integers
+        num_labels : int
+            number of unique objects
+        """
+        output = np.zeros_like(image)
+        for i, v in enumerate(np.unique(image)):
```

Comment:
also, [here](https://stackoverflow.com/questions/16992713/translate-every-element-in-numpy-array-according-to-key)'s some people benchmarking alternate solutions.  the vectorize solution might be applicable to our use case.

Reply:
Here's my translation of the vectorized version for our use case. 

```python3
In [11]: def relabel_vectorize(image):
    ...:     keys = np.unique(image)
    ...:     mapping = dict(zip(keys, range(len(keys))))
    ...:     return np.vectorize(mapping.__getitem__)(image)
```

and mine: 
```python3
In [10]: def relabel_image(image: np.ndarray) -> np.ndarray:
    ...:     output = np.empty_like(image)
    ...:     for i, v in enumerate(np.unique(image)):
    ...:         output[np.where(image == v)] = i
    ...:     return output, i
```

Tested against my implementation on a poisson-generated random data
```python
In [15]: a = np.random.poisson(0.5, size=1000000).reshape(1000, 1000)

In [16]: %timeit relabel_image(a)
62.9 ms ± 823 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [17]: %timeit relabel_vectorize(a)
191 ms ± 3.56 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [18]: a = np.random.poisson(0.5, size=10000000).reshape(1000, 10000)

In [19]: %timeit relabel_image(a)
694 ms ± 16.8 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [20]: %timeit relabel_vectorize(a)
1.97 s ± 16.9 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

The latter size is similar to our standard case, so I think we can safely stick with the loop case (which is also clearer to understand, I think).